packages/template-retail-react-app/app/components/store-locator-modal/index.jsx

```diff
@@ -24,70 +26,159 @@ import {
 } from '@salesforce/retail-react-app/app/constants'
 import {noop} from '@salesforce/retail-react-app/app/utils/utils'
 import {useForm} from 'react-hook-form'
+import {isServer} from '@salesforce/retail-react-app/app/utils/utils'
 
-const StoreLocatorModal = ({isOpen, onClose = noop}) => {
+const StoreLocatorModal = ({onClose = noop}) => {
     const intl = useIntl()
-
-    const [searchStoresParams, setSearchStoresParams] = useState({
+    var defaultSearchStoresParams = {
         countryCode: DEFAULT_STORE_LOCATOR_COUNTRY_CODE,
-        postalCode: DEFAULT_STORE_LOCATOR_POSTAL_CODE
-    })
+        postalCode: DEFAULT_STORE_LOCATOR_POSTAL_CODE,
+        limit: 10
+    }
+    var defaultUserHasSetManualGeolocation = false
+    if (!isServer) {
+        var searchStoresParamsFromLocalStorage = JSON.parse(
+            window.localStorage.getItem('STORE_LOCATOR_SEARCH_STORES_PARAMS')
+        )
+        var userHasSetManualGeolocationFromLocalStorage = window.localStorage.getItem(
+            'STORE_LOCATOR_USER_HAS_SET_GEOLOCATION'
+        )
+        if (searchStoresParamsFromLocalStorage)
+            defaultSearchStoresParams = searchStoresParamsFromLocalStorage
+        if (userHasSetManualGeolocationFromLocalStorage)
+            defaultUserHasSetManualGeolocation = Boolean(
+                userHasSetManualGeolocationFromLocalStorage
+            )
+    }
+    const [userHasSetManualGeolocation, setUserHasSetManualGeolocation] = useState(
+        defaultUserHasSetManualGeolocation
+    )
 
-    const form = useForm({
-        mode: 'onChange',
-        reValidateMode: 'onChange',
-        defaultValues: {
-            countryCode: searchStoresParams.countryCode,
-            postalCode: searchStoresParams.postalCode
-        }
+    const [searchStoresParams, setSearchStoresParams] = useState(defaultSearchStoresParams)
+    const searchStoresDataRef = useRef({
+        data: []
     })
```

Comment:
Using useRef to remember the last store results. Having this prevents the stores list from jumping to the top when the "Load More" button is clicked